airbnb_project/listings/harvester_app/harvester/spiders/listings_spider.py

```diff
@@ -0,0 +1,312 @@
+                        cord in
+                        coordinates]
+        except FileNotFoundError:
+            print("The file was not found")
+        except json.JSONDecodeError:
+            print("Error decoding JSON")
+        return requests
+
+    async def parse(self, response: Response, **kwargs):
+        """
+        Parse method for extracting listings from the Airbnb search results page.
+
+        Args:
+            response (Response): The response object from the request.
+
+        Yields:
+            Request: A request object for each listing's details page.
+        """
+        if response.status != 200:
+            return
+        script_tag = response.css(f'script#{SCRIPT_TAG}')
+        script_inner_text = script_tag.css('script::text').get()
+        script_json = json.loads(script_inner_text)
+        results = self._parse_listings_json(script_json)
+        current_url = response.url
+
+        if ListingsSpider.next_page_cursors is None:
+            ListingsSpider.next_page_cursors = self._get_cursors(script_json)
+
+        for result in results:
+            try:
+                listing = result.get("listing", {})
+                listing_id: str = listing.get("id")
+                print("listing_id", listing_id)
+                title = listing.get("title", "")
+                name = listing.get("name", "")
+                coordinate = listing.get("coordinate", {})
+                latitude = coordinate.get("latitude", "")
+                longitude = coordinate.get("longitude", "")
+                room_type = listing.get("roomTypeCategory", "")
+                page_url = API_CALL.format(n(t("StayListing", listing_id)))
+                airbnb_params = {
+                    "airbnb_listing_id": listing_id,
+                    "title": title,
+                    "name": name,
+                    "latitude": latitude,
+                    "longitude": longitude,
+                    "room_type": room_type
+                }
+
+                # A second request is made to get the registration number and other information using the listing_id
+                # which is scarped
+                if listing_id and listing_id not in ListingsSpider.ALREADY_SCRAPED_LISTINGS:
+                    yield Request(url=page_url, callback=self.handle_listing,
+                                  headers={'X-Airbnb-Api-Key': API_KEY},
+                                  meta={'airbnb_params': airbnb_params})
+                else:
+                    print("duplicate", listing_id)
+
+            except Exception as e:
+                print("error_listing", result)
+                print("exception", e.args[0])
+
+        if len(ListingsSpider.next_page_cursors) != 0:
+            cursor_id = ListingsSpider.next_page_cursors.pop()
+            next_url = f'{current_url}&cursor={cursor_id}'
+            yield response.follow(next_url, callback=self.parse)
```

Comment:
is this a Request object, similar to line 162?

Reply:
The `response.follow()` method returns a request object. It is useful in this context because we are navigating within the same relative URL as the initial response, simply changing the cursor ID to paginate through the result